### PR TITLE
Query: Caching: Use Microsoft.Framework.Cache.Memory as underlying query...

### PR DESCRIPTION
**Pull request by anpete**
... cache storage. Improved the parameterizer to support composite member access expressions.

### CODE REVIEW

File: src/EntityFramework.Core/EntityFramework.Core.csproj [hunk 1]
```diff
@@ -47,6 +47,9 @@
     <PackageReference Include="System.Collections.Immutable">
       <TargetFramework>portable-net45+win8+wp8+wpa81</TargetFramework>
     </PackageReference>
+    <PackageReference Include="Microsoft.Framework.Cache.Memory">
+      <TargetFramework>net45</TargetFramework>
+    </PackageReference>
```
Comment (bricelam) [inline src/EntityFramework.Core/EntityFramework.Core.csproj]: Do you get any build warnings because of this?

Comment (anpete) [inline src/EntityFramework.Core/EntityFramework.Core.csproj]: Nada

File: src/EntityFramework.Core/Query/CompiledQueryCache.cs [hunk 2]
```diff
@@ -30,14 +30,18 @@ public class CompiledQueryCache : ICompiledQueryCache
...
-            public bool IsAsync;
-            public IModel Model;
         }
 
-        private readonly ICollection<CompiledQuery> _items = new List<CompiledQuery>();
+        private readonly IMemoryCache _memoryCache;
+
+        public CompiledQueryCache([NotNull] IMemoryCache memoryCache)
+        {
+            Check.NotNull(memoryCache, "memoryCache");
```
Comment (bricelam) [inline src/EntityFramework.Core/Query/CompiledQueryCache.cs]: `nameof`

Comment (anpete) [inline src/EntityFramework.Core/Query/CompiledQueryCache.cs]: Going to wait until R# 9.1 is stable.

**Comment by bricelam** [comment]
:shipit:

**Comment by anpete** [comment]
Merged

**Pull request closed without merging**